modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/components/AutoSearch.tsx

```diff
@@ -0,0 +1,65 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {ClayButtonWithIcon} from '@clayui/button';
+import {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import React, {useState} from 'react';
+
+interface ISearchInterface {
+	onSearch: Function;
+	query: string;
+}
+
+const AutoSearch = ({onSearch, query}: ISearchInterface) => {
+	const [focused, setFocused] = useState<boolean>(false);
+
+	return (
+		<ClayInput.Group>
+			<ClayInput.GroupItem>
+				{!focused && (
+					<ClayInput.GroupInsetItem before tag="span">
+						<ClayIcon
+							className="inline-item inline-item-before"
+							focusable="false"
+							role="presentation"
+							symbol="search"
+						/>
+					</ClayInput.GroupInsetItem>
+				)}
+
+				<ClayInput
+					insetAfter={Boolean(focused)}
```

Comment:
Why do you need `Boolean` here? We already know that `focused` is and always will be a boolean because TS shows that.